TypeCobol/Compiler/Sql/Scanner/SqlScanner.cs

```diff
@@ -50,46 +80,242 @@ public override Token GetTokenStartingFrom(int startIndex)
+                case 'X':
+                case 'x':
+                    if ((startIndex < lastIndex - 1)&& (line[startIndex + 1] == '\'' || line[startIndex + 1] == '"'))
+                    {
+                        return ScanAlphanumericLiteral(startIndex+1, TokenType.HexadecimalAlphanumericLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanUserDefinedWord(startIndex);
+                    }
+                case 'B':
+                case 'b':
+                    if ((startIndex + 2 <= lastIndex) && (line[startIndex + 1] == 'x' || line[startIndex + 1] == 'X') &&
+                        (line[startIndex + 2] == '\'' || line[startIndex + 2] == '"'))
+                    {
+                        return ScanAlphanumericLiteral(startIndex, TokenType.SQL_BinaryStringLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanUserDefinedWord(startIndex);
+                    }
+                case 'U':
+                case 'u':
+                case 'G':
+                case 'g':
+                    if ((startIndex + 2 <= lastIndex) && (line[startIndex + 1] == 'x' || line[startIndex + 1] == 'X') &&
+                        (line[startIndex + 2] == '\'' || line[startIndex + 2] == '"'))
+                    {
+                        return ScanAlphanumericLiteral(startIndex, TokenType.SQL_GraphicStringLiteral, null);
+                    }
+                    else
+                    {
+                        return ScanUserDefinedWord(startIndex);
+                    }
+                default:
+                    return ScanUserDefinedWord(startIndex);
 
             }
+        }
 
-            if (IsSqlKeywordPart(line[currentIndex]))
+        private Token ScanSqlNumericLiteral(int startIndex, bool hasExplicitSign)
+        {
+            if (TryScanDecimalFloatingPointSpecialValue(startIndex, hasExplicitSign, out var decimalFloatingPointLiteralTokenValue))
             {
-                //Consume all sql-keyword compatible chars
-                for (; currentIndex <= lastIndex && IsSqlKeywordPart(line[currentIndex]); currentIndex++){}
-                string tokenText = line.Substring(startIndex, currentIndex - startIndex);
-                //Try to match keyword text
-                var tokenType = TokenUtils.GetSqlKeywordTokenTypeFromTokenString(tokenText);
-                switch (tokenType)
+                Token decimalFloatingPointSpecialValueToken = new Token(TokenType.SQL_DecimalFloatingPointLiteral,
+                    startIndex, currentIndex - 1, tokensLine)
+                {
+                    LiteralValue = decimalFloatingPointLiteralTokenValue
+                };
+                return decimalFloatingPointSpecialValueToken;
+            }
+
+            var token = ScanNumericLiteral(startIndex, '.');
```

Comment:
Put the '.' into a constant or add a comment to explain that SQL always uses dot as decimal point.

`private const char SQL_DECIMAL_POINT = '.';`